### PR TITLE
[documentation] #42: Restructure the tutorial

### LINKED ISSUE #42: Restructure the tutorial
- [ ] Improve overall tutorial structure without altering the contents of the individual topics.

My notes regarding this:

1. Some sections should be renamed (e.g., "conclusions"), some information should be moved around and grouped differently presenting the contents of the tutorial in a logical way.

2. Most of the info in Appendix shouldn't be there at all, because it's not something additional to the tutorial, it's the essential part of the tutorial. This might be a question of renaming the chapter and splitting it in a one or two other chapters.

    - My proposal is to have a "Learn the concepts" type of chapter: e.g. the concept of "genesis block" -- it's mentioned in lang-specific guides, but its only described in appendix and its definition is never referenced from the lang-specific guides. This might not be essential for people already familiar with blockchains but the tutorial is aiming at all kinds of audiences.
    - Config files better be described in something like "Configurations" section.
    - Why is this info https://hyperledger.github.io/iroha-2-docs/guide/appendix/running-iroha-on-bare-metal.html is in the appendix? It's just a more advanced topic than the lang-specific ones, people won't go looking in the appendix for this information.

3. Address this question: do we think there are a lot of people who are interested in iroha 1 vs. iroha 2 comparison *before* they learn what this tutorial can offer them? My proposal is to highlight this comparison but not put it at the very beginning.

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

- Split the appendix into two different sections: first covers configuration/management, and the second is "advanced mode" (running Iroha on bare metal). No changes to the content itself.
- Moved some of the info around:
  - Prerequisites from Intro -> to "Build and install" as the first section there
  - Conclusions -> to the Intro (renamed to "Tutorial updates"), rephrased a little, removed outdated info
  - Iroha 1 vs Iroha 2 -> into a separate topic, added sub-headers (no changes to the content itself)
  - info about ISI from the intro to Advanced topics -> into a separate file (no changes to the content itself)
  - info about Triggers from the intro to Advanced topics -> added to the existing topic on Triggers (no changes to the content itself)
- Added navigation section to the introduction to hopefully help users a bit with the understanding of what's going on in each section of the tutorial

Closes #42 and partially addresses #43 and #45 as well

<details> <summary>TOC screenshot</summary>
image
</details>

### CODE REVIEW

File: .vitepress/config.ts [hunk 1]
```diff
@@ -30,12 +30,16 @@ function getNav() {
 function getGuideSidebar() {
   return [
     {
-      text: 'Getting started',
+      text: 'Get started',
```
Comment (appetrosyan) [inline .vitepress/config.ts:33]: This is a common expression. 

I agree that this is grammatically dubious, but this is what people normally expect.

File: .vitepress/config.ts [hunk 2]
```diff
@@ -86,39 +90,43 @@ function getGuideSidebar() {
         {
          text: 'Queries',
          link: '/guide/advanced/queries',
-        },
+        }
```
Comment (appetrosyan) [inline .vitepress/config.ts:93]:
```suggestion
        },
```

A trailing comma is considered a good practice in TypeScript and in Rust.

File: .vitepress/config.ts [hunk 3]
```diff
@@ -86,39 +90,43 @@ function getGuideSidebar() {
...
           text: 'Genesis Block',
-          link: '/guide/appendix/genesis',
+          link: '/guide/configure/genesis',
         },
         {
           text: 'Keys',
-          link: "/guide/appendix/keys",
-        },
+          link: "/guide/configure/keys",
+        },        {
```
Comment (appetrosyan) [inline .vitepress/config.ts:115]:
```suggestion
        },        
        {
```

File: .vitepress/config.ts [hunk 4]
```diff
@@ -86,39 +90,43 @@ function getGuideSidebar() {
...
+      ]
+    },
+    {
+      text: 'Advanced Mode',
+      children: [
         {
-          text: 'Running Iroha On Bare Metal',
-          link: '/guide/appendix/running-iroha-on-bare-metal',
-        },
+          text: 'Run Iroha On Bare Metal',
```
Comment (appetrosyan) [inline .vitepress/config.ts:125]:
```suggestion
          text: 'Iroha On Bare Metal',
```

File: src/guide/advanced/running-iroha-on-bare-metal.md [hunk 5]
```diff
@@ -1,19 +1,20 @@
-# Running Iroha on bare metal
+# Run Iroha on bare metal
```
Comment (appetrosyan) [inline src/guide/advanced/running-iroha-on-bare-metal.md:1]:
```suggestion
# Iroha on bare metal
```

File: src/guide/advanced/triggers.md [hunk 6]
```diff
@@ -1,6 +1,30 @@
...
+
+The documentation on the `EventFilter` types is under construction, as we
+are likely to make major changes to that particular architecture. For now,
+suffice it to say that you can look at the source code in
+`iroha_data_model` and see a few particularly interesting applications.
+
+## Triggers in More Detail
 
-# Triggers in More Detail
+<!-- We should probably have a Queen of Hearts here, as we have a lot of `execute`, `trigger` and `block` -->
```
Comment (appetrosyan) [inline src/guide/advanced/triggers.md:27]: :)

File: src/guide/bash.md [hunk 7]
```diff
@@ -53,7 +53,7 @@ is phased out and `iroha_python` is the official scripting interface, we
 might use the excellent Python libraries for creating a useful TUI. For
 now, one must make do with what one has.
 
-## 1. Iroha 2 Client Setup
+## 1. Set up Iroha 2 Client
```
Comment (appetrosyan) [inline src/guide/bash.md:56]:
```suggestion
## 1. How to set up an Iroha 2 Client
```
Alternatively, the continuous verb forms in headings are common, especially in software documentation. 

I think we need to have a **brief** discussion about this.

File: src/guide/build-and-install.md [hunk 8]
```diff
@@ -1,5 +1,23 @@
 # Build and Install Iroha 2
 
+## Prerequisites
+
+For this tutorial, you will need:
+
+- [git](https://githowto.com/)
+- [A working Rust toolchain](https://www.rust-lang.org/learn/get-started):
+  cargo, rust v1.60 and up [^1]
```
Comment (appetrosyan) [inline src/guide/build-and-install.md:9]:
```suggestion
  `cargo`, `rustc` v1.60 and up [^1]
```

File: src/guide/build-and-install.md [hunk 9]
```diff
@@ -1,5 +1,23 @@
...
+- (Optional) [Docker](https://docs.docker.com/get-docker/)
+- (Optional) [Docker compose](https://docs.docker.com/compose/) [^2]
+
+[^1]:
+    If you’re having issues with installing rust compatible with our code
```
Comment (appetrosyan) [inline src/guide/build-and-install.md:14]:
```suggestion
    If you’re having issues with installing Rust compatible with our code
```

File: src/index.md [hunk 10]
```diff
@@ -3,111 +3,22 @@
...
-
-## Tutorial preamble
+## Preamble
 
-What follows is an introduction suitable for both experienced developers,
-prospective users, and people casually curious about blockchain technology.
-The level of detail is sufficient so that you wouldn't need any
-supplementary guide. However, should you want to learn more, we have more
-detailed documentation in the works.
+The tutorial is suitable for both experienced developers, prospective
```
Comment (appetrosyan) [inline src/index.md:17]:
```suggestion
This tutorial is suitable for both experienced developers, prospective
```

File: src/index.md [hunk 11]
```diff
@@ -117,33 +28,62 @@ In this guide, we shall
...
-The _appendix_ contains a variety of useful information. In particular, the
-three main configuration files are explained: the _peer_ configuration and
-the _genesis block_, which you need to get right to start a network; and
-the _client_ configuration, which you adjust in order to interact with the
-blockchain.
+## Navigation
+
+We invite you to follow the tutorial in this order:
+
+1. If you are already familiar with Hyperledger Iroha, read about
```
Comment (appetrosyan) [inline src/index.md:35]:
```suggestion
0. If you are already familiar with Hyperledger Iroha, read about
```

File: .vitepress/config.ts (hunk 1 above)
Comment (outoftardis) [reply .vitepress/config.ts:33]: I'd advocate for a shorter form but I'm generally ok with both in this particular case

File: src/guide/bash.md (hunk 7 above)
Comment (outoftardis) [reply src/guide/bash.md:56]: in this case we have steps we expect the user to follow so the imperative works better than "how to"

File: src/index.md [hunk 12]
```diff
@@ -117,33 +28,62 @@ In this guide, we shall
...
+   [the differences](./guide/iroha-2.md) between two versions of Iroha.
+2. [Build and install Iroha 2](./guide/build-and-install.md), then follow
```
Comment (outoftardis) [inline src/index.md:37]:
```suggestion
1. [Build and install Iroha 2](./guide/build-and-install.md), then follow
```

File: src/index.md [hunk 13]
```diff
@@ -117,33 +28,62 @@ In this guide, we shall
...
+   one of the language-specific guides to learn how to set up and configure
+   Iroha 2, register a domain and an account, register and mind assets, and
+   visualize outputs:
+   - [Bash](./guide/bash.md)
+   - [Python](./guide/python.md)
+   - [Rust](./guide/rust.md)
+   - [Kotlin/Java](./guide/kotlin-java.md)
+   - [Javascript (TypeScript)](./guide/javascript.md)
+3. Dive into more advanced topics to deepen your understanding of Iroha 2:
```
Comment (outoftardis) [inline src/index.md:46]:
```suggestion
2. Dive into more advanced topics to deepen your understanding of Iroha 2:
```

File: src/index.md [hunk 14]
```diff
@@ -117,33 +28,62 @@ In this guide, we shall
...
+   - Check the
+     [features that make Iroha truly special](./guide/advanced/intro.md).
+   - Learn in greater detail about Iroha 2
+     [configuration and management](./guide/configure/intro.md).
+4. Use Iroha 2 in a more advanced mode:
```
Comment (outoftardis) [inline src/index.md:51]:
```suggestion
3. Use Iroha 2 in a more advanced mode:
```

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**